### PR TITLE
Header Checker: Misc Enhancements 

### PR DESCRIPTION
**Pull request by dachalco**
- Avoid checking deleted files from changelist. Specifically check {added, modified, renamed}.
- Add regex pattern for exclusion
- Update exclusion list per each repo. See `FREERTOS_IGNORED_*` AND `KERNEL_IGNORED_*` in the PR file changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/scripts/check-header.py [hunk 1]
```diff
@@ -105,76 +113,352 @@ def configArgParser():
                         help    = 'Treat arguments json files that store a list of files to check.')
     return parser
 
+#--------------------------------------------------------------------------------------------------
+#                                            CONFIG  
+#--------------------------------------------------------------------------------------------------
+FREERTOS_IGNORED_EXTENSIONS = [
+    '.1',
+    '.ASM',
+    '.C',
```
Comment (cobusve) [inline .github/scripts/check-header.py:122]: Is this right? I think cpp files should have the same header

File: .github/scripts/check-header.py [hunk 2]
```diff
@@ -105,76 +113,352 @@ def configArgParser():
...
+    '.DSW',
+    '.G_C',
+    '.H',
```
Comment (cobusve) [inline .github/scripts/check-header.py:125]: Same here

File: .github/scripts/check-header.py [hunk 3]
```diff
@@ -105,76 +113,352 @@ def configArgParser():
...
+    '.bts',
+    '.ccxml',
+    '.cdkproj',
+    '.cdkws',
+    '.cfg',
+    '.cgp',
+    '.cmake',
+    '.cmd',
+    '.config',
+    '.cpp',
```
Comment (cobusve) [inline .github/scripts/check-header.py:158]: and here

**Review by cobusve — APPROVED** [review]

File: .github/scripts/check-header.py (hunk 1 above)
Comment (dachalco) [reply .github/scripts/check-header.py:122]: It's correct for current state of repo in that no `FreeRTOS/FreeRTOS` files of these types include the header. These should be skipped. 

The changes in `header-checks.yml` now use the check as defined in the user PR so users can update as needed

**Pull request merged**